src/transformers/pipelines/question_answering.py

```diff
@@ -330,26 +338,41 @@ def preprocess(self, example, padding="do_not_pad", doc_stride=128, max_question
                         qas_id=None,
                     )
                 )
-        return {"features": features, "example": example}
+
+        shallow = []
```

Comment:
Why shallow for the name?

Reply:
Could be a better name, I just tried to disassemble the `feature` before passing it on for `_forward` as it makes it benefit automatically from `no_grad` and `to(device)` from `forward` (otherwise `_forward` had to take care of it).

Reusing `features` is good enough? Didn't find a better name.

Reply:
I like features personally (and I use example/sample when I want to make the distinction: one sample gives several features).